rasa/nlu/selectors/response_selector.py

```diff
@@ -642,8 +633,41 @@ def load(
 
         return model
 
+    def _uses_sequence_features_for_text(self) -> bool:
+        """Whether we make use of sequence features for the TEXT attribute.
+
+        Note that, just like the DIETClassifier, DIET2BOW can make use of sentence
+        features even if the number of transformer layers is set to 0 because it creates
+        a BOW representation from sequence+sentence features.
+
+        In contrast to that, DIET2DIET classifier uses the last token from the
+        sequence that is obtained by concatenating sequence and sentence features.
+        Hence, it only makes use of sequence features if and only if the number of
+        transformer layers is > 0.
+        """
+        return (
+            self.component_config[NUM_TRANSFORMER_LAYERS] > 0
+            or not self.use_text_as_label
+        )
+
+    def _needs_sentence_features_for_labels(self) -> bool:
```

Comment:
Just thought you had found a bug. I agree that is unnecessarily confusing. Will rename that :)

Reply:
While I was about to change it, I remembered why that was named like this. - The difference was that we always expect sequence features but not always sentence features. E.g. there are sometimes fallbacks for sentence features which are build from sequence features. Here, there's a fallback to multihot sentence features instead - which is pretty arbitrary and should better be sum of sequence features as well...
I think, It wasn't handled in the previous version at all and I think there is only one featurizer that might do that.

Reply:
> It wasn't handled in the previous version at all and I think there is only one featurizer that might do that.

This might be related to my previous question: Which one featurizer forces us to maintain this piece of code here? 😅

Reply:
My hunch is `LexicalSyntacticFeaturizer`. Is that it?
If so, I highly doubt any configuration will contain `LexicalSyntacticFeaturizer` alone.

Reply:
Yes, that one in particular. And in general, that no Featurizer gives me any guarantees on what it adds.

> If so, I highly doubt any configuration will contain LexicalSyntacticFeaturizer alone.
yeah but we're fixing edge cases here anyway, so why not catch that one as well :)

Reply:
I get that but do we need to support the edge case like this and bloat up our code? Shouldn't we then raise a proper exception saying that they didn't add any featurizer which would compute sentence level features?
Looking through the list of featurizers, I can't see any other featurizer which would not compute sentence level features.

Reply:
>  Shouldn't we then raise a proper exception saying that they didn't add any featurizer which would compute sentence level features

Yep, we can do that. And we can/do use the above function to raise that exception in that case, no? 

Also, for me it made the code a bit easier to understand, so I would vote for keeping those _use* and _need* function as they helped me clarifying what the architectures will do.

Reply:
resolving this - Johannes raised same question